moneta/src/python/customauth/forms.py

```diff
@@ -0,0 +1,19 @@
+from django import forms
+from django.contrib.auth import authenticate
+
+
+class UserLoginForm(forms.Form):
+    username = forms.CharField(widget=forms.TextInput)
+    password = forms.CharField(widget=forms.PasswordInput)
+
+    def clean(self, *args, **kwargs):
+        username = self.cleaned_data.get('username')
+        password = self.cleaned_data.get('password')
+        if username and password:
+            user = authenticate(username=username, password=password)
+            if not user:
+                raise forms.ValidationError('This user does not exist')
+            if not user.check_password(password):
+                raise forms.ValidationError('Incorrect password!')
```

Comment:
No !.

Reply:
Which algorithm must I use for hashing a password?
Because before checking the password, got from the user with existing password in the database, I must hash got password and after that compare these 2 passwords